### PR TITLE
Adding C++ module file extensions and implementing "allmodulespublic"

### PR DESCRIPTION
**Pull request by hannes-harnisch**
See #1773. That PR was closed accidentally due to reshuffling commits on request.

### CODE REVIEW

**Comment by hannes-harnisch** [comment]
@starkos Is this okay?

File: modules/vstudio/vs2010_vcxproj.lua [hunk 1]
```diff
@@ -1511,6 +1512,18 @@
 		end
 	end
 
+	function m.allModulesPublic(cfg)
+		if _ACTION >= "vs2019" then
+			if cfg.allmodulespublic ~= nil then
+				if cfg.allmodulespublic then
+					m.element("AllProjectBMIsArePublic", nil, "true")
```
Comment (samsinsane) [inline modules/vstudio/vs2010_vcxproj.lua:1519]: For future reference, you can use `iif(<cond>, "true", "false")` instead of duplicating the line.

**Review by samsinsane — APPROVED** [review]

File: modules/vstudio/vs2010_vcxproj.lua (hunk 1 above)
Comment (hannes-harnisch) [reply modules/vstudio/vs2010_vcxproj.lua:1519]: Which line?

Comment (nickclark2016) [reply modules/vstudio/vs2010_vcxproj.lua:1523]: To follow up, that would take the place of the if-else block. It'd be:

```lua
m.element("AllProjectBMIsArePublic", nil, iif(cfg.allmodulespublic, "true", "false"))
```

**Pull request merged**